libpromises/evalfunction.c

```diff
@@ -248,10 +248,11 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             convert = JsonArrayCreate(RlistLen(value));
             for (const Rlist *rp = value; rp != NULL; rp = rp->next)
             {
-                if (rp->val.type == RVAL_TYPE_SCALAR &&
-                    strcmp(RlistScalarValue(rp), CF_NULL_VALUE) != 0)
+                char* item = RlistScalarValueSafe(rp);
```

Comment:
I'm removing the _safe_ part here as we want it to crash if it turns out not being a SCALAR after all (programming error indeed).

Hmmm on a second look I'm removing the whole `if` clause, since I don't want any special handling for `cf_null`.